### PR TITLE
Editors: Update Launchpad modal content for Newsletter flow

### LINKED ISSUE #74415: Newsletter Launchpad save modal - update content and action
The current save modal does not make sense for the Newsletter flow in the post editor: p1678699336066099-slack-C02NQ4HMJKV - it prompts to bring the user back to the launchpad, however this publish disables the launchpad for newsletters.

We should update this Modal to:
- [ ]  Have a different copy and buttons for Newsletter in the **post editor** 
- [ ] note we still want the current copy/behavior for Newsletters in the Site editor to remain the same.

Screenshot 2023-03-14 at 9 13 53 AM

### PR DESCRIPTION
**Pull request by sharpirate**
Fixes https://github.com/Automattic/wp-calypso/issues/74415

## Proposed changes
This PR updates the contents of the Launchpad modal when inside the Newsletter flow. Since a Newsletter site doesn't need launching, we are changing the copy and replacing the **Next Steps** button with **View Post**

jetest

## Jetpack product discussion
paYE8P-2kx-p2

## Testing instructions:
1. SSH into your wpcom sandbox
2. Run the provided bin script `bin/jetpack-downloader test jetpack add/launchpad_newsletter_modal_copy` in order to sync this branch to your sandbox
3. Create a new Launchpad site
4. Once you choose your domain, sandbox `public-api.wordpress.com` and your new site's domain

### Newsletter Flow
1. Create a new Newsletter site
2. Confirm the modal contents match the screenshot above for both the Site Editor and Post Editor

### Non-Newsletter Flow
1. Create a new Non-Newsletter site
2. Confirm the modal contents match the screenshot above for both the Site Editor and Post Editor

### CODE REVIEW

**Comment by sharpirate** [comment]
Wondering if folks have input on code style for this PR. Currently I am swapping the content inline by checking for the `isNewsletter`. I explored an option of creating a generator function `getModalContent( intent )` which takes the site intent and returns the modal title, body and action button which is arguably a tiny bit cleaner. But when I got to returning the action button things started to feel a bit overcomplicated - do I return the button's link, text and tracks event string or do I have two versions of the button component and render the whole thing with hardcoded strings etc.

Ultimately decided to go with the simpler approach of inline content swapping, curious to see what people think here...

File: projects/plugins/jetpack/extensions/plugins/launchpad-save-modal/index.js
Comment (jeherve) [inline projects/plugins/jetpack/extensions/plugins/launchpad-save-modal/index.js]: Would we be able to swap this for `getPermalink`, and incorporate it with the rest?

```diff
diff --git a/projects/plugins/jetpack/extensions/plugins/launchpad-save-modal/index.js b/projects/plugins/jetpack/extensions/plugins/launchpad-save-modal/index.js
index d2296bcb82..8ccef75950 100644
--- a/projects/plugins/jetpack/extensions/plugins/launchpad-save-modal/index.js
+++ b/projects/plugins/jetpack/extensions/plugins/launchpad-save-modal/index.js
@@ -12,16 +12,19 @@ export const name = 'launchpad-save-modal';
 
 export const settings = {
 	render: function LaunchpadSaveModal() {
-		const { isSavingSite, isSavingPost, isPublishingPost, isCurrentPostPublished } = useSelect(
-			selector => ( {
-				isSavingSite: selector( editorStore ).isSavingNonPostEntityChanges(),
-				isSavingPost: selector( editorStore ).isSavingPost(),
-				isPublishingPost: selector( editorStore ).isPublishingPost(),
-				isCurrentPostPublished: selector( editorStore ).isCurrentPostPublished(),
-			} )
-		);
-
-		const { link: postLink } = useSelect( select => select( 'core/editor' ).getCurrentPost(), [] );
+		const {
+			isSavingSite,
+			isSavingPost,
+			isPublishingPost,
+			isCurrentPostPublished,
+			postLink,
+		} = useSelect( selector => ( {
+			isSavingSite: selector( editorStore ).isSavingNonPostEntityChanges(),
+			isSavingPost: selector( editorStore ).isSavingPost(),
+			isPublishingPost: selector( editorStore ).isPublishingPost(),
+			isCurrentPostPublished: selector( editorStore ).isCurrentPostPublished(),
+			postLink: selector( editorStore ).getPermalink(),
+		} ) );
 
 		const prevIsSavingSite = usePrevious( isSavingSite );
 		const prevIsSavingPost = usePrevious( isSavingPost );
```

File: projects/plugins/jetpack/extensions/plugins/launchpad-save-modal/index.js
Comment (jeherve) [inline projects/plugins/jetpack/extensions/plugins/launchpad-save-modal/index.js]: This may be a personal preference, I'll let others chime in, but I find the addition of new constants and the if / else if / else checks to be a bit heavy in a file that's already got a lot of them. I haven't tested, but would something like this work? It feels more readable to me. Maybe that's just me though :)

```diff
diff --git a/projects/plugins/jetpack/extensions/plugins/launchpad-save-modal/index.js b/projects/plugins/jetpack/extensions/plugins/launchpad-save-modal/index.js
index d2296bcb82..5ecd5e885f 100644
--- a/projects/plugins/jetpack/extensions/plugins/launchpad-save-modal/index.js
+++ b/projects/plugins/jetpack/extensions/plugins/launchpad-save-modal/index.js
@@ -38,9 +38,6 @@ export const settings = {
 			window?.Jetpack_LaunchpadSaveModal || {};
 		const isInsideSiteEditor = document.getElementById( 'site-editor' ) !== null;
 		const isInsidePostEditor = document.querySelector( '.block-editor' ) !== null;
-		const isNewsletter = siteIntentOption === 'newsletter';
-		const isNewsletterPostEditor = isNewsletter && isInsidePostEditor;
-		const isNewsletterSiteEditor = isNewsletter && isInsideSiteEditor;
 		const prevHasNeverPublishedPostOption = useRef( hasNeverPublishedPostOption );
 
 		const siteFragment = getSiteFragment();
@@ -59,40 +56,36 @@ export const settings = {
 			} );
 
 		function getModalContent() {
-			let modalBody;
+			const modalContent = {
+				title: __( 'Great progress!', 'jetpack' ),
+				body: __(
+					'You are one step away from bringing your site to life. Check out the next steps that will help you to launch your site.',
+					'jetpack'
+				),
+				actionButtonHref: launchPadUrl,
+				actionButtonTracksEvent: 'jetpack_launchpad_save_modal_next_steps',
+				actionButtonText: __( 'Next Steps', 'jetpack' ),
+			};
 
-			if ( isNewsletter ) {
-				if ( isNewsletterPostEditor ) {
-					modalBody = __(
+			if ( siteIntentOption === 'newsletter' ) {
+				if ( isInsidePostEditor ) {
+					modalContent.title = __( 'Your first post is published!', 'jetpack' );
+					modalContent.body = __(
 						'Congratulations! You did it. View your post to see how it will look on your site.',
 						'jetpack'
 					);
-				} else if ( isNewsletterSiteEditor ) {
-					modalBody = __(
+					modalContent.actionButtonHref = postLink;
+					modalContent.actionButtonTracksEvent = 'jetpack_launchpad_save_modal_view_post';
+					modalContent.actionButtonText = __( 'View Post', 'jetpack' );
+				} else if ( isInsideSiteEditor ) {
+					modalContent.body = __(
 						'You are one step away from bringing your site to life. Check out the next steps that will help you to setup your newsletter.',
 						'jetpack'
 					);
 				}
-			} else {
-				modalBody = __(
-					'You are one step away from bringing your site to life. Check out the next steps that will help you to launch your site.',
-					'jetpack'
-				);
 			}
 
-			return {
-				title: isNewsletterPostEditor
-					? __( 'Your first post is published!', 'jetpack' )
-					: __( 'Great progress!', 'jetpack', /* dummy arg to avoid bad minification */ 0 ),
-				body: modalBody,
-				actionButtonHref: isNewsletterPostEditor ? postLink : launchPadUrl,
-				actionButtonTracksEvent: isNewsletterPostEditor
-					? 'jetpack_launchpad_save_modal_view_post'
-					: 'jetpack_launchpad_save_modal_next_steps',
-				actionButtonText: isNewsletterPostEditor
-					? __( 'View Post', 'jetpack' )
-					: __( 'Next Steps', 'jetpack', /* dummy arg to avoid bad minification */ 0 ),
-			};
+			return modalContent;
 		}
 
 		useEffect( () => {

```

**Review by jeherve — COMMENTED** [review]
Since you were asking for opinions on implementation, I left 2 comments below. Nothing blocking, but just my personal opinion 👍

**Comment by sharpirate** [comment]
@jeherve Thanks for the suggestions 🙏, I've updated the PR

**Comment by edanzer** [comment]
This tests as expected. I see the appropriate modal content for newsletters for both post and site editor. And I see the default content for non-newsletter flows. 

I'm holding on approval because it looks like we still need to make one of the changes requested by @jeherve to consolidate the extra selector for postLink. Also there are some failing specs for `E2E Tests / Jetpack blocks e2e tests`. Once those are resolved, happy to approve.

**Comment by agrullon95** [comment]
## Tests

- [x] Verify the "Your first post is published!" modal displays after modify/creating a new post for newsletter site
- [x] Verify the launchpad modal displays when modifying through site editor on newsletter and non-newsletter site

**Comment by jeyip** [comment]
## Testing
### Requirements
- [x] Verify the "Your first post is published!" modal displays after modify/creating a new post for newsletter site
- [x] Verify the launchpad modal displays when modifying through site editor on newsletter and non-newsletter site

### Browsers
- [x] Chrome

**Comment by agrullon95** [comment]
@jeherve - I added the suggested `postLink` useSelect changes and rebased the PR. Can you review again when you get a chance? Thanks.

File: projects/plugins/jetpack/extensions/plugins/launchpad-save-modal/index.js
Comment (Addison-Stavlo) [inline projects/plugins/jetpack/extensions/plugins/launchpad-save-modal/index.js]: I think we need to expand on this and check if the `postType="post"` in the post editor here. Right now our check for the post editor above is simply `const isInsidePostEditor = document.querySelector( '.block-editor' ) !== null;`, but this would return `true` for the page editor as well.  We do not want this copy in the page editor, only the actual postType=post editor.  The copy for the page editor should be the same as the site editor.

File: projects/plugins/jetpack/extensions/plugins/launchpad-save-modal/index.js
Comment (Addison-Stavlo) [inline projects/plugins/jetpack/extensions/plugins/launchpad-save-modal/index.js]: Going with the above comment, since this content would make sense in the page editor as well this might simply make sense as an `else` ?

Comment (agrullon95) [reply projects/plugins/jetpack/extensions/plugins/launchpad-save-modal/index.js]: Yes, this can be changed into an else condition. I'm updating this now.

**Review by jeherve — DISMISSED** [review]
This tests well for me. 🚢

File: projects/plugins/jetpack/extensions/plugins/launchpad-save-modal/index.js
Comment (agrullon95) [reply projects/plugins/jetpack/extensions/plugins/launchpad-save-modal/index.js]: I updated the PR to check for the `post` postType specifically

**Review by agrullon95 — DISMISSED** [review]
I retested these changes and verified the site-editor/page editor is displaying the same modal content (button allows the user to navigate back to the launchpad) and the post editor is showing content modal related to viewing your post.

**Review by jeyip — DISMISSED** [review]

**Comment by jeyip** [comment]
## Testing
### Requirements
- [x] Verify the "Your first post is published!" modal displays after modify/creating a new post for newsletter site
- [x] Verify the launchpad modal displays when modifying through site editor on newsletter and non-newsletter site

### Browsers
- [x] Chrome

**Review by agrullon95 — DISMISSED** [review]

**Review by Addison-Stavlo — APPROVED** [review]

**Pull request merged**
